vip-plugins/vip-plugins.php

```diff
@@ -0,0 +1,222 @@
+		}
+		$actions['vip-code-activated-plugin'] = __( 'Enabled via code', 'vip-dashboard' );
+
+		if ( 'plugins' === $screen->id ) {
+			unset( $actions['network_active'] );
+		}
+	}
+
+	return $actions;
+}
+add_filter( 'plugin_action_links', 'wpcom_vip_plugin_action_links', 10, 4 );
+add_filter( 'network_admin_plugin_action_links', 'wpcom_vip_plugin_action_links', 10, 4 );
+
+/**
+ * Merge code activated plugins with database option for better UI experience
+ *
+ * @param  array $value
+ * @param  string $option
+ * @return array
+ */
+function wpcom_vip_option_active_plugins( $value, $option ) {
+	$code_plugins = wpcom_vip_get_filtered_loaded_plugins();
+	$value = array_merge( $code_plugins, $value );
+
+	return $value;
+}
+add_filter( 'option_active_plugins', 'wpcom_vip_option_active_plugins', 10, 2 );
+
+/**
+ * Merge code activated plugins with network database option for better UI experience
+ *
+ * @param  array $value
+ * @param  string $option
+ * @return array
+ */
+function wpcom_vip_site_option_active_sitewide_plugins( $value, $option ) {
+	$code_plugins = wpcom_vip_get_network_filtered_loaded_plugins();
+	$value = array_merge( $code_plugins, $value );
+
+	return $value;
+
+}
+add_filter( 'site_option_active_sitewide_plugins', 'wpcom_vip_site_option_active_sitewide_plugins', 10, 2 );
+
+/**
+ * Unmerge code activated plugins from active plugins option (reverse of the above)
+ *
+ * @param  array $value
+ * @param  array $old_value
+ * @param  string $option
+ * @return array
+ */
+function wpcom_vip_pre_update_option_active_plugins( $value, $old_value, $option ) {
+	$code_plugins = wpcom_vip_get_filtered_loaded_plugins();
+	$value = array_diff( $value, $code_plugins );
+
+	return $value;
+}
+add_filter( 'pre_update_option_active_plugins', 'wpcom_vip_pre_update_option_active_plugins', 10, 3 );
+
+/**
+ * Unmerge code activated plugins from network active plugins option (reverse of the above)
+ *
+ * @param  array $value
+ * @param  array $old_value
+ * @param  string $option
+ * @param  int $network_id
+ * @return array
+ */
+function wpcom_vip_pre_update_site_option_active_sitewide_plugins( $value, $old_value, $option, $network_id ) {
+	$code_plugins = wpcom_vip_get_network_filtered_loaded_plugins();
+	$value = array_diff( $value, $code_plugins );
+
+	return $value;
+}
+add_filter( 'pre_update_site_option_active_sitewide_plugins', 'wpcom_vip_pre_update_site_option_active_sitewide_plugins', 10, 4 );
+
+/**
+ * Custom CSS and JS for the plugins UIs
+ *
+ * @return null
+ */
+function wpcom_vip_plugins_ui_admin_enqueue_scripts() {
+	$screen = get_current_screen();
+	if ( 'plugins' === $screen->id || 'plugins-network' === $screen->id ) {
+		wp_enqueue_style( 'vip-plugins-style', WP_CONTENT_URL . '/mu-plugins/' . basename( __DIR__ ) . '/css/plugins-ui.css' , '3.0' );
+		wp_enqueue_script( 'vip-plugins-script', WP_CONTENT_URL . '/mu-plugins/' . basename( __DIR__ ) . '/js/plugins-ui.js', array( 'jquery' ), '3.0', true );
```

Comment:
We can simplify the URL-building: `plugins_url( 'css/plugins-ui.css', __FILE__ )`

Reply:
Good call, didn't realised this worked for mu-plugins too